os/pm/pm_checkstate.c

```diff
@@ -127,37 +120,37 @@ enum pm_state_e pm_checkstate(int domain)
 	 */
 
 	now = clock_systimer();
-	if (now - pdom->stime >= TIME_SLICE_TICKS) {
+	if (now - g_pmglobals.stime >= TIME_SLICE_TICKS) {
 		int16_t accum;
 
 		/* Sample the count, reset the time and count, and assess the PM
 		 * state.  This is an atomic operation because interrupts are
 		 * still disabled.
 		 */
 
-		accum       = pdom->accum;
-		pdom->stime = now;
-		pdom->accum = 0;
+		accum       = g_pmglobals.accum;
+		g_pmglobals.stime = now;
+		g_pmglobals.accum = 0;
 
 		/* Reassessing the PM state may require some computation.  However,
 		 * the work will actually be performed on a worker thread at a user-
 		 * controlled priority.
 		 */
 
-		(void)pm_update(domain, accum);
+		(void)pm_update(accum);
 	}
 
 	/* Consider the possible power state lock here */
 
-	for (index = 0; index < pdom->recommended; index++) {
-		if (pdom->stay[index] != 0) {
-			pdom->recommended = index;
+	for (index = 0; index < CONFIG_PM_NDOMAINS; index++) {
+		if (g_pmglobals.suspend_count[index] != 0) {
+			g_pmglobals.recommended = PM_STANDBY;
```

Comment:
if IDLE domain or LCD domain were called suspend, pm state should be PM_NORMAL

Reply:
For LCD case, we will create a different PR